src/jit/importer.cpp

```diff
@@ -4283,262 +4283,12 @@ NamedIntrinsic Compiler::lookupNamedIntrinsic(CORINFO_METHOD_HANDLE method)
-                            else if (strcmp(methodName, "Int64") == 0)
-                            {
-                                result = NI_Base_Vector256_AsInt64;
-                            }
-                            else if (strcmp(methodName, "SByte") == 0)
-                            {
-                                result = NI_Base_Vector256_AsSByte;
-                            }
-                            else if (strcmp(methodName, "Single") == 0)
-                            {
-                                result = NI_Base_Vector256_AsSingle;
-                            }
-                            else if (strcmp(methodName, "UInt16") == 0)
-                            {
-                                result = NI_Base_Vector256_AsUInt16;
-                            }
-                            else if (strcmp(methodName, "UInt32") == 0)
-                            {
-                                result = NI_Base_Vector256_AsUInt32;
-                            }
-                            else if (strcmp(methodName, "UInt64") == 0)
-                            {
-                                result = NI_Base_Vector256_AsUInt64;
-                            }
-                        }
-                        else if (strcmp(methodName, "CreateScalarUnsafe") == 0)
-                        {
-                            result = NI_Base_Vector256_CreateScalarUnsafe;
-                        }
-                        else if (strcmp(methodName, "GetElement") == 0)
-                        {
-                            result = NI_Base_Vector256_GetElement;
-                        }
-                        else if (strcmp(methodName, "GetLower") == 0)
-                        {
-                            result = NI_Base_Vector256_GetLower;
-                        }
-                        else if (strcmp(methodName, "ToScalar") == 0)
-                        {
-                            result = NI_Base_Vector256_ToScalar;
-                        }
-                        else if (strcmp(methodName, "WithElement") == 0)
-                        {
-                            result = NI_Base_Vector256_WithElement;
-                        }
-                    }
-                    else if (strcmp(className, "`1") == 0)
-                    {
-                        if (strcmp(methodName, "get_Zero") == 0)
-                        {
-                            result = NI_Base_Vector256_Zero;
-                        }
-                    }
-                }
-#endif // _TARGET_XARCH_
-            }
-        }
 #if defined(_TARGET_XARCH_)
-        else if (strcmp(namespaceName, ".X86") == 0)
-        {
-            result = HWIntrinsicInfo::lookupId(className, methodName, enclosingClassName);
-        }
+        assert(strcmp(namespaceName + 25, ".X86") == 0 || namespaceName[25] == '\0');
+        result = HWIntrinsicInfo::lookupId(className, methodName, enclosingClassName);
```

Comment:
@CarolEidt, is this going to play well with the need for all `IsSupported` checks to be done (i.e. for `Arm64` intrinsics to explicitly return false on x86).

Reply:
I think that will require some separate work. I don't think that the VM even communicates the non-platform intrinsics to the JIT, so changes will be needed there.

Reply:
> (i.e. for Arm64 intrinsics to explicitly return false on x86).

This is controlled by managed code, e.g., `Avx.PlatformNotSupported.cs`.

Reply:
@fiigii, there is an issue (https://github.com/dotnet/coreclr/issues/20379) tracking us needing to recognize the `IsSupported` flag for all ISAs, even those on unsupported platforms, to ensure that the constant folding always happens (otherwise it is dependent on several other JIT optimizations, like inlining, successfully happening).

Reply:
Right. What's awkward is that we don't want `System.Runtime.Intrinsics.Arm.Arm64.Simd` to be marked as `[Intrinsic]`, but we would like `System.Runtime.Intrinsics.Arm.Arm64.Simd.IsSupported` to be. I'm not sure whether that would be an issue.

Reply:
Got it, thanks for explaining.